JP-3342: Reorganize outlier detection documentation (#8880)
spacetelescope/jwst: docs/jwst/outlier_detection/outlier_detection_ifu.rst

.. _outlier-detection-ifu:

Integral Field Unit (IFU) Data
==============================

This module serves as the interface for applying ``outlier_detection`` to IFU
observations, like those taken with NIRSpec and MIRI. A :ref:`Stage 3 association <asn-level3-techspecs>`,
which is loaded into a :py:class:`~jwst.datamodels.ModelContainer` object,
serves as the basic format for all processing performed by this step.
The basis of the outlier detection flagging for IFU data  is to look for pixels on the detector
that are regularly discrepant from their neighbors, with a sharper division than could be explained
by the detector PSF.
This routine performs the following operations:

> #. Extracts parameter settings from input ModelContainer and merges them with any user-provided values.

Reviewer: ```suggestion
#. Extract parameter settings for the input ModelContainer and merge them with any user-provided values.
```

   See :ref:`outlier detection arguments <outlier_detection_step_args>` for the full list of parameters.

> #. Loops over cal files, computing nearest-neighbor differences for each pixel

Reviewer: ```suggestion
#. Loop over cal files, computing nearest-neighbor differences for each pixel
```

   in the along-dispersion direction.
   For MIRI, with the dispersion axis along the y axis, the neighbors that are used to
   to find the differences are to the left and right of each pixel being examined.
   For NIRSpec, with the dispersion along the x axis, the neighbors that are used to
   find the differences are above and below the pixel being examined.
   The smaller of the two (left, right or up, down) differences is stored as the difference value for each
   pixel. This avoids artifacts from bright edges.

> #. Compares the nearest-neighbor differences across science exposures to find the minimum

Reviewer: ```suggestion
#. Compare the nearest-neighbor differences across science exposures to find the minimum
```

   neighbor difference at each detector pixel.

> #. Determines a local spatial median of the minimum difference array using a median filter with a kernel size

Reviewer: ```suggestion
#. Determine a local spatial median of the minimum difference array using a median filter with a kernel size
```

   set by the user according to the ``kernel_size`` parameter.

> #. Normalizes the minimum difference array by the local median.

Reviewer: ```suggestion
#. Normalize the minimum difference array by the local median.
```

> #. Selects outliers by flagging those normalized minimum values larger than the ``threshold_percent``

Reviewer: ```suggestion
#. Select outliers by flagging those normalized minimum values larger than the ``threshold_percent``
```

   parameter.

> #. Updates input ImageModel DQ arrays with mask of detected outliers.

Reviewer: ```suggestion
#. Update input ImageModel DQ arrays with mask of detected outliers.
```

Reviewer: Also - SCI, ERR, and VAR arrays are also updated.

> After launch it was discovered that the bad pixels on the MIRI detectors vary with time.

Reviewer: This paragraph should probably go at the top, before going into the detailed procedure.
